### PR TITLE
Use Apache HTTP Client

### PR DESCRIPTION
**Pull request by ctienshi**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: src/main/java/org/wso2/callhome/internal/CallHome.java [hunk 1]
```diff
@@ -306,52 +318,51 @@ private URL constructCallHomeURL(ExtractedInfo extractedInfo) throws CallHomeExc
...
+        SSLContextBuilder sslContextBuilder = SSLContexts.custom();
+
+        try {
+            sslContextBuilder = sslContextBuilder.loadTrustMaterial(trustStoreFile, trustStorePassword.toCharArray());
+            SSLContext sslContext = sslContextBuilder.build();
+            SSLConnectionSocketFactory sslConSocFactory = new SSLConnectionSocketFactory(sslContext,
+                    new NoopHostnameVerifier());
+            HttpClientBuilder httpClientBuilder = HttpClients.custom();
+            httpClientBuilder.setSSLSocketFactory(sslConSocFactory);
+
```
Comment (kasunsiyambalapitiya) [inline src/main/java/org/wso2/callhome/internal/CallHome.java]: Shall we remove the empty line here?

File: src/main/java/org/wso2/callhome/internal/CallHome.java [hunk 2]
```diff
@@ -364,65 +374,41 @@ private String getResponse(HttpsURLConnection httpsURLConnection) throws CallHom
...
-                        log.debug(url + " Gateway timeout");
-                        break;
-                    case HttpURLConnection.HTTP_UNAVAILABLE:
-                        log.debug(url + " Unavailable");
-                        break;
-                    default:
-                        log.debug(url + " Unknown response code");
-                        break;
+        HttpGet request = new HttpGet(String.valueOf(url));
+
```
Comment (kasunsiyambalapitiya) [inline src/main/java/org/wso2/callhome/internal/CallHome.java]: Shall we remove the empty line here as well?

File: src/main/java/org/wso2/callhome/internal/CallHome.java (hunk 1 above)
Comment (ctienshi) [reply src/main/java/org/wso2/callhome/internal/CallHome.java]: Fixed

File: src/main/java/org/wso2/callhome/internal/CallHome.java (hunk 2 above)
Comment (ctienshi) [reply src/main/java/org/wso2/callhome/internal/CallHome.java]: Fixed

**Pull request merged**